plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -2843,4 +2844,22 @@ public function update_meta( &$object, $meta ) {
 
 		return $update_meta;
 	}
+
+	/**
+	 * Perform after meta change operations, including updating the date_modified field, clearing caches and applying changes.
+	 *
+	 * @param WC_Abstract_Order $order Order object.
+	 * @param \WC_Meta_Data     $meta  Metadata object.
+	 */
+	protected function after_meta_change( &$order, $meta ) {
+		$current_date_time = new \WC_DateTime( current_time( 'mysql', 1 ), new \DateTimeZone( 'GMT' ) );
+		method_exists( $meta, 'apply_changes' ) && $meta->apply_changes();
+		$this->clear_caches( $order );
+
+		// Prevent this happening multiple time in same request.
+		if ( $order->get_date_modified() < $current_date_time && empty( $order->get_changes() ) ) {
+			$order->set_date_modified( current_time( 'mysql' ) );
```

Comment:
I was thinking that the call to `current_time` here should also have the `1` gmt switch, like above, but I guess that `set_date_modified` handles that already?